docker/new-api/Dockerfile

```diff
@@ -17,8 +18,13 @@ COPY --chown=node:node curriculum/ curriculum/
 COPY --chown=node:node client/ client/
 
 RUN pnpm config set dedupe-peer-dependents false
+# While we want to ignore scripts generally, we do need to generate the prisma
+# client. Note: npx is the simplest way to generate the client without us having
+# to have prisma as a prod dependency. The jq tricks are to ensure we're using
+# the right version of prisma.
 RUN pnpm install -F=api -F=curriculum -F tools/scripts/build -F challenge-parser \
-  --frozen-lockfile
+  --frozen-lockfile --ignore-scripts
+RUN cd api && npx prisma@$(jq -r '.devDependencies.prisma' < package.json) generate
```

Comment:
Out of curiosity why are we using `npx` for this and not `pnpm dlx`?

Reply:
To be honest there doesn't seem to be much difference. Ironically, npx is a hair faster, but there's not much in it. Also it's shorter, which I like!

TL;DR: personal preference.